pilot/pkg/networking/core/v1alpha3/configgen.go

```diff
@@ -42,6 +43,9 @@ func NewConfigGenerator(plugins []plugin.Plugin) *ConfigGeneratorImpl {
 }
 
 func (configgen *ConfigGeneratorImpl) BuildSharedPushState(env *model.Environment, push *model.PushContext) error {
+	if !pilot.EnableCDSPrecomputation() {
```

Comment:
I am worried about the latency when we have large number of proxies， like 10k 。

Reply:
About the latency, we must test in large scale with comparison.

Reply:
the latency increase in serving cds to proxies does not matter that much.. it might increase by 1ms, but this is control plane info. At runtime, the proxy is still using final computed information. At 10k proxies, we would have atleast 10+ pilots. Imagine every pilot computing every possible CDS output and storing it all in local pilot memory (gigabytes of memory for each pilot instance). It will quickly become unmanageable and unstable